### ISSUE TITLE
Plan for Foundry 0.8.x compatibility

### ISSUE THREAD

**Issue by javierriveracastro** [javierriveracastro/betteroll-swade#126]
I'm opening this bug to try to plan for a simple transition with the less possible pain.

For now, it looks like a total API overhaul. Having a version that works both with 0.7 and 0.8 looks quite difficult.

The plan for now:

1. Test SWADE system with the alpha when it is released. SWADE doesn't even load.
2. Test Betterrolls with the alpha when possible.

**Comment by Razortide** [comment]
Hello Javier,
as a non coder - what can I do to help?

**Comment by javierriveracastro** [comment]
Testing. I will try to upload something as soon as possible.

The first plan is releasing a version compatible with both 0.7 and 0.8. So I need people testing it in both versions.

**Comment by Razortide** [comment]
Why not going all in for 0.8? Doesn't this will save you time?
I am happy to test as soon as you have a version.

**Comment by javierriveracastro** [comment]
Because lots of systems are not working with 0.7 and a sizeable part of the users is unlikely to update soon.

That said, if compatibility is too hard to maintain I'll drop it.

**Comment by Razortide** [comment]
Got it. I was just thinking: You *have* a working 0.7.10. version, therefore a "dual-compatible" version looks like a lot of work to me ;-)

**Comment by javierriveracastro** [comment]
For now, in programming time, close to no effort. Things can change if I find some unexpected problem, of course.

Problem here is testing time, it is just double.

**Comment by SalieriC** [comment]
Don't even try to talk javier out of it. ^^'

I'll gladly (keept to) test BR2 on 0.7.10.

**Comment by richardgaywood** [comment]
I'll be very happy to contribute to the testing effort.

**Comment by themadphoenix** [comment]
I can help with testing.  I have both Forge and a local server that I test various modules on.

**Comment by themadphoenix** [comment]
But, as an issue with the 8.6 version of Foundry/Forge, the buttons on the settings do nothing if you click them.  Also, I can still drag something out and drop it on a token and it targets, but it doesn't roll.  I'll verify that tonight, though.

**Comment by Razortide** [comment]
v.0.8.6:

As discussed: No weapon card rolled from an actor (or token) sheet works.
Here is how I setup the test instance:
1. Copy Data folder from productive (0.7.10) instance (Raspberry)
2. Installed 0.8.6 on local windows machine
3. Copied data folder into local environment
4. Started Foundry
5. Updated systems
5. Started World (received an error that migration of the world might have issues because the Foundry version (0.7.10) is too old for a frictionless migration)

When I roll a skill that works just fine (without Elan, but that might be my bad) and factors in the modifiers.

I found that I am unable to open the "World Global Actions". I receive the following error:
![image](https://user-images.githubusercontent.com/35796457/120559432-4ebdb900-c401-11eb-92ab-1cc6c60b73a6.png)

Or optional rules:
![image](https://user-images.githubusercontent.com/35796457/120559534-80cf1b00-c401-11eb-92dc-f3d91712fbc7.png)

When I drag and drop a weapon to another token I receive this error
![image](https://user-images.githubusercontent.com/35796457/120559792-f3d89180-c401-11eb-94cf-f881547a6f62.png)

Clicking on the weapon will trigger this error:
![image](https://user-images.githubusercontent.com/35796457/120559856-14085080-c402-11eb-9243-fd270b68f2bf.png)

**Comment by javierriveracastro** [comment]
< #229 > should be part of this issue.

**Comment by javierriveracastro** [comment]
Please, all you, if you can, try this update:

https://www.dropbox.com/s/xv838rya7fyqlc5/betterrolls-swade2.zip?dl=1

It should be version 2.33

**Comment by Razortide** [comment]
I can confirm that now all reported errors in https://github.com/javierriveracastro/betteroll-swade/issues/126#issuecomment-853422961 have been fixed.

I've found, that now all weapons (including melee weapons) trigger the "Ammo deduction". I am using German terms, so I am not sure if this is the reason.
![2021-06-03_21h57_18](https://user-images.githubusercontent.com/35796457/120704267-ae29d080-c4b6-11eb-8cfc-94191b8c2e6f.jpg)

I can reproduce this with the "Drag & Drop" attack, but not with a click on the quick access item (although the target remains the same). Range between tokens seem to be irrelevant.

EDIT2: This bug also appears in V2.31 and 0.7.10
